src/metrics/documentation_updated/index.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  SignedDataEvent,
+  MetricsSignature,
+  MetricData,
+} from "../../interfaces";
+import {
+  DocumentationUpdatedOutput,
+  DocumentationUpdatedPayload,
+} from "./interfaces";
+import octokit from "../../core/octokit";
+import { logger } from "../../core/logger";
+
+export const collectDocumentationUpdatedMetrics = async (
+  dataEvent: SignedDataEvent
+): Promise<MetricData> => {
+  const payload = dataEvent.payload as DocumentationUpdatedPayload;
+
+  const repo = payload.repository.name;
+  const owner = payload.repository.owner.login;
+  const number = payload.number;
+
+  let mdFileNames: string[] = [];
+
+  try {
+    const request = await octokit.request(
+      "GET /repos/{owner}/{repo}/pulls/{pull_number}/commits",
+      {
+        owner: owner,
+        repo: repo,
+        pull_number: payload.pull_request.number,
+      }
+    );
+
+    const data = request.data;
+
+    const commitShas = data.map((commit) => commit.sha);
+
+    for (let i = 0; i < commitShas.length; i++) {
+      const commitSha = commitShas[i];
+      const commitRequest = await octokit.request(
+        "GET /repos/{owner}/{repo}/commits/{ref}",
+        {
+          owner: owner,
+          repo: repo,
+          ref: commitSha,
+        }
+      );
```

Comment:
Even though we might have less resource issues in the future, I'd like to be careful with doing requests in a loop here. For all we know someone is merging 500 commits (e.g. merging the hyperactive main branch back into their age-old feature branch).

I thought about maybe using the .diff URL the PR event gives us, but this endpoint seems even more fitting to just get a list of the PR-affected files in a single request: 
`/repos/{owner}/{repo}/pulls/{pull_number}/files`

https://docs.github.com/en/free-pro-team@latest/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests-files

Let's change it to that one, or does that not work for some reason?

Reply:
That sounds good. I just didn't find this endpoint when looking for a way to get the files. Then I also don't have to get rid of duplicates. Thank you.